concrete/src/Entity/File/Version.php

```diff
@@ -509,85 +704,84 @@ public function getExtension()
+     * Get the the tags assigned to the file version (as a list of strings).
      *
-     * @return Version
+     * @return string[]
      */
-    public function duplicate()
+    public function getTagsList()
     {
-        $em = \ORM::entityManager();
-        $qq = $em->createQuery('SELECT max(v.fvID) FROM \Concrete\Core\Entity\File\Version v where v.file = :file');
-        $qq->setParameter('file', $this->file);
-        $fvID = $qq->getSingleScalarResult();
-        ++$fvID;
-
-        $fv = clone $this;
-        $fv->fvID = $fvID;
-        $fv->fvIsApproved = false;
-        $fv->fvDateAdded = new \DateTime();
-        $uID = (int) (new User())->getUserID();
-        if ($uID !== 0) {
-            $fv->fvAuthorUID = $uID;
-        }
-
-        $em->persist($fv);
-
-        $this->deny();
-
-        foreach ($this->getAttributes() as $value) {
-            $value = clone $value;
-            /*
-             * @var $value AttributeValue
-             */
-            $value->setVersion($fv);
-            $em->persist($value);
+        $tags = explode("\n", str_replace("\r", "\n", trim($this->getTags())));
+        $clean_tags = [];
+        foreach ($tags as $tag) {
+            $tag = trim($tag);
+            if ($tag !== '') {
+                $clean_tags[] = $tag;
+            }
         }
 
-        $em->flush();
-
-        $fe = new \Concrete\Core\File\Event\FileVersion($fv);
-        Events::dispatch('on_file_version_duplicate', $fe);
-
-        return $fv;
+        return $clean_tags;
     }
 
     /**
-     * Deny a file version update.
+     * Get the the tags assigned to the file version (one tag per line - lines are separated by '\n').
+     *
+     * @return null|string
      */
-    public function deny()
+    public function getTags()
     {
-        $this->fvIsApproved = false;
-        $this->save();
-        $fe = new \Concrete\Core\File\Event\FileVersion($this);
-        Events::dispatch('on_file_version_deny', $fe);
+        return $this->fvTags;
     }
 
     /**
-     * Save changes to a file.
+     * Get the mime type of the file if known.
      *
-     * @param bool $flush Flush the EM cache
+     * @return string|false
      */
-    protected function save($flush = true)
+    public function getMimeType()
     {
-        $em = \ORM::entityManager();
-        $em->persist($this);
-        if ($flush) {
-            $em->flush();
+        try {
+            $fre = $this->getFileResource();
+            $result = $fre->getMimetype();
+        } catch (FileNotFoundException $x) {
+            $result = false;
         }
+
+        return $result;
+    }
+
+    /**
+     * Get the type of the file.
+     *
+     * @return \Concrete\Core\File\Type\Type
+     */
+    public function getTypeObject()
+    {
+        $app = Application::getFacadeApplication();
+        $fh = $app->make('helper/file');
+        $ext = $fh->getExtension($this->fvFilename);
+        $ftl = FileTypeList::getType($ext);
```

Comment:
`return FileTypeList::getType($ext);` ?

Reply:
I just kept the existing code, and IMHO it's better to keep different lines for different actions...